www/casestudies/items/endtoend/authoring.php

```diff
@@ -0,0 +1,154 @@
+$signedRequest = $Init->generate();
+
+?>
+
+<div class="jumbotron section">       
+    <div class="overview">
+        <h1>End to End Demo – Authoring</h1>
+        <p>Learnosity's Author API enable professional authors (as well as teachers) to create and edit content.</p>
+        <p>Questions are saved to the Learnosity item bank to be included in the assessment.</p>
+    </div>
+</div>
+
+<div class="section">        
+    <div class="row">
+        <div id="notifications" class="col-md-2"></div>
+        <div class="col-md-10">
+            <div id="learnosity-author"></div>
+        </div>
+    </div>  
+    <p class="text-right">
+        <br>
+        <a class="btn btn-primary btn-md" href="javascript:addNextQuestion();">Add more</a>
+        <a class="btn btn-primary btn-md" href="javascript:goToAssessment();">Go to Assessment</a>
+    </p>
+</div>
+
+<script src="//authorapi.learnosity.com?v0.10"></script>
+<script>
+    var initOptions = <?php echo $signedRequest; ?>;
+    var itemIDs = new Array();
+
+    var authorApp = LearnosityAuthor.init(initOptions, {
+
+        readyListener: function () {            
+            authorApp.on('save:success', function (event) {
+                saveItemID(initOptions.request.reference);
+            });
+        }
+
+    });
+
+    function showNotification (itemID) {            
+        var $message = $('<a/>').text('Question ' + itemIDs.indexOf(itemID))
+                                .attr('href','javascript:editItem("' + itemID + '")')
+        var $closeBtn = $('<button/>').attr('type', 'button')
+                                      .attr('data-dismiss', 'alert')
+                                      .attr('aria-hidden', 'true')
+                                      .attr('title', 'Delete question')
+                                      .attr('onclick', 'javascript:removeItem("' + itemID + '")')
+                                      .addClass('close')
+                                      .text('×');
+        var $notification = $('<div/>').addClass('alert alert-info alert-dismissable')
+                                       .attr('id', 'ItemNotification' + itemID)
+                                       .append($closeBtn)
+                                       .append($message);
+        $('#notifications').append($notification);
+    }
+
+    function addNextQuestion() {
+        initOptions.request.reference = guid();
+        authorApp.setItem(initOptions.request.reference);
+    }
+
+    function goToAssessment() {
+        window.location = 'assessment.php?itemIDs=' + itemIDs.join(",");
+    }
+
+    function guid() {
+      function s4() {
+        return Math.floor((1 + Math.random()) * 0x10000)
+          .toString(16)
+          .substring(1);
+      }
+      return s4() + s4() + '-' + s4() + '-' + s4() + '-' +
+        s4() + '-' + s4() + s4() + s4();
+    }
+
+    function saveItemID(itemID) {
+        var idx = itemIDs.indexOf(itemID);
```

Comment:
you can use Jquery and use .inArray() instead to simplify this check, and keep it in the "if" statement